cmd/internal/database/postgres/upgrade.go

```diff
@@ -0,0 +1,241 @@
+package postgres
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const (
+	postgresHBAConf     = "pg_hba.conf"
+	postgresConfigCmd   = "pg_config"
+	postgresUpgradeCmd  = "pg_upgrade"
+	postgresInitDBCmd   = "initdb"
+	postgresVersionFile = "PG_VERSION"
+	oldPostgresBindir   = "/usr/local/bin/pg-old"
+)
+
+// Upgrade indicates whether the database files are from a previous version of and need to be upgraded.
```

Comment:
```suggestion
// Upgrade performs an upgrade of the database in case a newer version of the database is detected.
//
// The function aborts the update without returning an error as long as the old data stays unmodified and only prints out the error to console. This behavior is intended to reduce unnecessary downtime caused by misconfigurations.
```